.github/workflows/nightly-t5x-test-mgmn.yaml

```diff
@@ -66,45 +66,9 @@ jobs:
       EXPERIMENT_SUBDIR: T5X
     secrets: inherit
 
-  publish-completion:
-    needs: [metadata, run-jobs]
-    uses: ./.github/workflows/_publish_badge.yaml
-    if: success() || failure()
-    secrets: inherit
-    with:
-      ENDPOINT_FILENAME: 't5x-test-overall-status.json'
-      PUBLISH: ${{ github.event_name == 'workflow_run' || needs.metadata.outputs.PUBLISH == 'true' }}
-      SCRIPT: |
-        STATUS=failure
-        if [[ ${{ needs.run-jobs.result }} == "success" ]]; then
-          EXIT_STATUSES="${GITHUB_RUN_ID}-*/*-status.json"
-          PASSED_TESTS=$(jq -r '. | select ((.state == "COMPLETED") and (.exitcode == "0")) | .state' $EXIT_STATUSES | wc -l)
-          FAILED_TESTS=$(jq -r '. | select ((.state != "COMPLETED") or (.exitcode != "0")) | .state' $EXIT_STATUSES | wc -l)
-          TOTAL_TESTS=$(ls $EXIT_STATUSES | wc -l)
-
-          echo "Test statuses:"
-          jq -rc 'input_filename,.' $EXIT_STATUSES
-
-          if [[ $FAILED_TESTS -eq 0 ]] && [[ $TOTAL_TESTS -gt 0 ]] || [[ $PASSED_TESTS -eq $TOTAL_TESTS ]]; then
-            BADGE_COLOR=brightgreen
-            STATUS=success
-          elif [[ $PASSED_TESTS -eq 0 ]]; then
-            BADGE_COLOR=red
-          else
-            BADGE_COLOR=yellow
-          fi
-          echo "MESSAGE='${PASSED_TESTS}/${TOTAL_TESTS} passed'" >> $GITHUB_OUTPUT
-          echo "COLOR='${BADGE_COLOR}'" >> $GITHUB_OUTPUT
-        else
-          echo "MESSAGE='n/a'" >> $GITHUB_OUTPUT
-          echo "COLOR='red'" >> $GITHUB_OUTPUT
-        fi
-        echo "LABEL='Completion'" >> $GITHUB_OUTPUT
-        echo "STATUS='$STATUS'" >> $GITHUB_OUTPUT
-
   publish-verified:
-    if: needs.publish-completion.outputs.STATUS == 'success' && ((github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') || (github.event_name == 'workflow_dispatch' && inputs.PUBLISH))
-    needs: [metadata, publish-completion]
+    if: needs.run-jobs.outputs.TEST_STATUS == 'success' && ((github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') || (github.event_name == 'workflow_dispatch' && inputs.PUBLISH))
```

Comment:
Please move the logic to decide whether an image should be published to a separate `metadata` job. This would allow for better readability and the use of more complicated logics. For reference, see:
- https://github.com/NVIDIA/JAX-Toolbox/blob/1cec8a38ecc0ceb7f63b30ced4480910192fee7b/.github/workflows/nightly-jax-build.yaml#L59
- https://github.com/NVIDIA/JAX-Toolbox/blob/1cec8a38ecc0ceb7f63b30ced4480910192fee7b/.github/workflows/nightly-jax-build.yaml#L22

Reply:
The logic to publish is dependent on the test result, so it cannot be moved to an ancestor of the test job